project Testing merge request

Mikearaya opened the pull request:
I have tested all the public methods of each class for successful and failing operations.
test file are found inside `spec/`:
- `player_spec.rb`
- `board_spec.rb`
- `player_spec.rb`


### I'm kindly requesting your review before I can merge to development and master branch

**Thank you** :bowing_man:

Mikearaya committed:
Test player.rb

Mikearaya committed:
Test Board class

Mikearaya committed:
Update board class test

Mikearaya committed:
Test TicTacToe class

Mikearaya committed:
Update README file

Mikearaya committed:
Update package.json

Mikearaya committed:
Update description

Mikearaya committed:
Add Fuubar test formatter

AdnanAfsari reviewed (changes requested):
Status: Changes Required

Hello @Mikearaya ✋ 

Nice work 👏  but

- [ ] I was not able to run test in my local machine instead getting those errors.Please fix that.
<img width="747" alt="Screenshot 2020-05-08 at 9 26 38 PM" src="https://user-images.githubusercontent.com/8613710/81421059-ae03b800-9172-11ea-8ab7-5c2a1a0da606.png">

- [x] You should have one `it` block for each expected behaviour [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png). In some cases there are more than one.

Please go ahead and make those changes and submit for another review.

Happy Coding
Adnan Afsari

Mikearaya committed:
Remove duplicate expectation

Mikearaya commented:
Hi, @AdnanAfsari  Thank you for trying to fix the OS issue as much as you can.

the code runs on my side which I'm using Linux
![Screenshot from 2020-05-08 16-18-14](https://user-images.githubusercontent.com/9751246/81427478-88bb7d80-9163-11ea-951a-897e4540ce5b.png)

but it is raising the issue on @AdnanAfsari who was using Mac. I don't know if it is OS related or not but I could not duplicate the same error on my side.

other than that I have removed all the duplicate expect assertions from my spec. I added them to prove the same thing in reverse order false & true

### I'm kindly requesting for another review

**Thank you** :bowing_man:

mosesogwo reviewed (approved):
**HI @Mikearaya, good work with your tests:+1:. Your project is approved.**

You should add a test for **won?** to check that it doesn't return true when there are more than 5 moves, and there is no winning play yet.

**Good luck in your next project:stars:**
### Status: Approved:heavy_check_mark:
